docs/user/reference/hook.md

```diff
@@ -221,12 +221,16 @@ In normal operation, a unit will run at least the `install`, `start`, `config-ch
 
 When a charm is first deployed, the following hooks are executed in order before a charm reaches its operation phase:
 
-* storage-attached (if the unit has storage)
+* storage-attached (if the unit has storage) [1]
 * install
 * leader-elected (if the unit is a leader)
 * config-changed
 * start
 
+```{note}
+[1] Only machine charms have the behaviour where the `storage-attached` hook must run before the `install` hook. 
+```
+
```

Comment:
I'd move this note right after the item it's about. But don't worry about it -- whichever way you get the info out is good at this point!

Reply:
I've improved the explanatory text and kept the bullet items simple